cpp/src/arrow/compute/exec/exec_plan.h

```diff
@@ -408,6 +408,54 @@ struct ARROW_EXPORT Declaration {
   std::string label;
 };
 
+struct ARROW_EXPORT QueryOptions {
+  QueryOptions();
+
+  /// \brief Should the plan use a legacy batching strategy
+  ///
+  /// This is currently in place only to support the Scanner::ToTable
+  /// method.  This method relies on batch indices from the scanner
+  /// remaining consistent.  This is impractical in the ExecPlan which
+  /// might slice batches as needed (e.g. for a join)
+  ///
+  /// However, it still works for simple plans and this is the only way
+  /// we have at the moment for maintaining implicit order.
+  bool use_legacy_batching;
```

Comment:
nit: `= false` like all the other fields?

Reply:
Done (also got rid of the explicit constructor)